examples/plugins/cli-cmd/hello_priority/hello_priority.py

```diff
@@ -0,0 +1,13 @@
+from avocado.core.extension_manager import PLUGIN_PRIORITY_HIGH
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLICmd
+
+
+class HelloWorld(CLICmd):
```

Comment:
A `CLICmd` example is not really the best option here, because of the way Avocado currently selects the plugin to be run based on the command given.

What I mean is, with the following changes:

```diff
diff --git a/examples/plugins/cli-cmd/hello/hello.py b/examples/plugins/cli-cmd/hello/hello.py
index c9373dda8..8d5c13509 100644
--- a/examples/plugins/cli-cmd/hello/hello.py
+++ b/examples/plugins/cli-cmd/hello/hello.py
@@ -1,3 +1,4 @@
+from avocado.core.extension_manager import PLUGIN_PRIORITY_LOW
 from avocado.core.output import LOG_UI
 from avocado.core.plugin_interfaces import CLICmd
 
@@ -6,6 +7,7 @@ class HelloWorld(CLICmd):
 
     name = 'hello'
     description = 'The classical Hello World! plugin example.'
+    priority = PLUGIN_PRIORITY_LOW
 
     def run(self, config):
         LOG_UI.info(self.description)
diff --git a/examples/plugins/cli-cmd/hello_priority/hello_priority.py b/examples/plugins/cli-cmd/hello_priority/hello_priority.py
index c1764716b..db15f3281 100644
--- a/examples/plugins/cli-cmd/hello_priority/hello_priority.py
+++ b/examples/plugins/cli-cmd/hello_priority/hello_priority.py
@@ -6,7 +6,7 @@ from avocado.core.plugin_interfaces import CLICmd
 class HelloWorld(CLICmd):
 
     name = 'hello'
-    description = 'The classical Hello World! plugin example.'
+    description = 'The classical Hello World! plugin example (HIGH PRIORITY)'
     priority = PLUGIN_PRIORITY_HIGH
 
     def run(self, config):
```

Try to install both `hello` and `hello_priority` and see which one gets executed.  A relevant piece of code is here: 

https://github.com/avocado-framework/avocado/blob/217896fd3ddb9c6c2d879b39533fbe08b6670322/avocado/core/app.py#L108

Now, we need to decide wether priorities *should* work for this type of plugin. IMO, it is confusing having plugins with the same type and same name, and distinguishing them based on priority, but there could use cases that I'm missing.